app/src/main/kotlin/com/github/gotify/messages/MessagesActivity.kt

```diff
@@ -242,12 +249,15 @@ internal class MessagesActivity :
         refreshAll.setOnClickListener { refreshAll() }
     }
 
-    override fun onBackPressed() {
-        if (binding.drawerLayout.isDrawerOpen(GravityCompat.START)) {
-            binding.drawerLayout.closeDrawer(GravityCompat.START)
-        } else {
-            super.onBackPressed()
+    private fun addBackPressCallback() {
+        onBackPressedCallback = object : OnBackPressedCallback(false) {
+            override fun handleOnBackPressed() {
+                if (binding.drawerLayout.isDrawerOpen(GravityCompat.START)) {
```

Comment:
Do we need this check here? I'd expect that the callback is only executed when it's enabled, and it's only enabled if the drawer is open.

Reply:
I don't exactly know. If in any circumstances it is called and the drawer is not open, would that matter?  
I suppose it doesn't throw an exception, at least according to the docs:  
https://developer.android.com/reference/androidx/drawerlayout/widget/DrawerLayout#closeDrawer(int)

If so, we don't need that here, true.

Reply:
yeah, the two extra lines don't really hurt the code-base, so it's probably not a big deal if they stay and aren't really needed.